syntax = "proto3";
package base;


message Empty {}

message ISIN {
  string isin = 1;
}

message StockPrice {
  double price = 1;
}

message Asset {
  string isin = 1;
  int32 amount = 2;
  double price = 3;
}
  
message Order {
  string token = 1;
  Asset asset = 2;
  OrderType type = 3;
  bool is_buy = 4;    
}

enum OrderType {
  UNSPECIFIED = 0;
  MARKET = 1;
  LIMIT = 2;
}

message CancelOrderRequest {
  string token = 1;
  uint64 order_id = 2;
}
